### ISSUE TITLE
"upstream changed" indicator for partitioned assets

### ISSUE THREAD

**Issue by sryza** [dagster-io/dagster#9374]
Possible behavior: A partitioned asset should say "Upstream changed" if the partitions upstream of any of its partitions have changed more recently than those partitions.

Requests:
- https://dagster.slack.com/archives/C01U954MEER/p1660565555051989?thread_ts=1660561165.304039&cid=C01U954MEER

**Comment by mkleinbort-ic** [comment]
Yes, this is a problem for us as well. Especially for the static partitions

From the dagster documentation: 

![image](https://github.com/dagster-io/dagster/assets/112868935/c5fd8537-b51d-48c7-9256-60650394fc3b)

If you imagine `ml_model_for_each_ocean` takes in a dataframe of ocean data, I'd want all the partitions to show as stale if the ocean data changes.

**Comment by mkleinbort-ic** [comment]
What's the status of this? It's been over 1y since the issue was opened.

**Comment by mkleinbort-ic** [comment]
Hello - following up on this. 

It seems the Freshness Policy can trigger the materialization of statically partitioned assets on the change of an upstream asset.

That said, this is hard to view in the UI and the "materialize stale or missing" does not catch these partitions.

**Comment by mkleinbort-ic** [comment]
I think there's only a few places where I'd like to see this added:

For me the most pressing is the asset sidebar when looking at a partitioned asset on the DAG

![image](https://github.com/dagster-io/dagster/assets/112868935/b508f6f1-e052-459c-847d-5187b60d9061)

Would be: 

On the "materialized changed and missing" button - as this would be very handy to just update the partitions
On the assets themselves (though this would require some UI design)

![image](https://github.com/dagster-io/dagster/assets/112868935/31b12d54-b38d-43c9-a54a-c2e28aa72798)

**Comment by smackesey** [comment]
Thanks for the suggestions @mkleinbort-ic. We are currently working on overhauling this entire system to be primarily focused on the most recent partition. I'll make the sure engs/designers  working on this see your feedback.
